jira_sync/jira_wrapper.py

```diff
@@ -17,57 +17,63 @@
 
 class JIRA:
     """
-    Class for interacting with JIRA API.
-
-    Attributes:
-      jira: Instance of the jira.JIRA object
-      project: Project to work with
-      issue_type: Default issue type for creating issues
-      project_states: List of states on the project in format {"state": "id"}.
-                      Example: {"NEW": "1"}
+    Class for interacting with the JIRA API.
+
+    :attribute jira: Instance of the jira.JIRA object
+    :attribute project: Project to work with
+    :attribute issue_type: Default issue type for creating issues
+    :attribute project_statuses: List of statuses on the project in the format
+                                 {"status": "id"}.
+                                 Example: {"NEW": "1"}
     """
 
-    jira: jira.client.JIRA
+    jira: jira.client.JIRA | None
     project: str
     issue_type: str
-    project_states: dict[Issue, dict[str, str]]
+    project_statuses: dict[Issue, dict[str, str]]
 
     def __init__(
         self,
-        url: str,
-        token: str,
+        url: str | None,
+        token: str | None,
         project: str,
         issue_type: str,
     ):
         """
-        Class constructor.
+        Object constructor.
```

Comment:
I always thought that Object is instance of Class and constructor is creating the instance as you can't call it on existing object.

Reply:
Should have changed it to e.g. “Initialize the instance.” instead.

The term “constructor” is a bit out of place in Python – the `__new__()` class method creates an object (can even make it of a different class, whee) and the `__init__()` instance method initializes the created object.

Reply:
I know that `constructor` is not really a correct term in Python, but I was learning OOP on Java.